crates/wasi-common/src/old/snapshot_0/sys/windows/hostcalls_impl/fs.rs

```diff
@@ -458,52 +459,51 @@ pub(crate) fn path_filestat_set_times(
 
 pub(crate) fn path_symlink(old_path: &str, resolved: PathGet) -> Result<()> {
     use std::os::windows::fs::{symlink_dir, symlink_file};
-    use winx::winerror::WinError;
 
     let old_path = concatenate(resolved.dirfd(), Path::new(old_path))?;
     let new_path = resolved.concatenate()?;
 
     // try creating a file symlink
-    symlink_file(&old_path, &new_path).or_else(|e| {
-        match e.raw_os_error() {
-            Some(e) => {
-                log::debug!("path_symlink at symlink_file error code={:?}", e);
-                match WinError::from_u32(e as u32) {
-                    WinError::ERROR_NOT_A_REPARSE_POINT => {
-                        // try creating a dir symlink instead
-                        symlink_dir(old_path, new_path).map_err(Into::into)
-                    }
-                    WinError::ERROR_ACCESS_DENIED => {
-                        // does the target exist?
-                        if new_path.exists() {
-                            Err(Error::EEXIST)
-                        } else {
-                            Err(WinError::ERROR_ACCESS_DENIED.into())
-                        }
+    let err = match symlink_file(&old_path, &new_path) {
```

Comment:
I'd say that `or_else` is more readable

Reply:
Hmm, I actually have to agree with @alexcrichton on this one, and prefer explicit `match`/`if let` with early returns.